supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/Preprocessor.ts

```diff
@@ -0,0 +1,314 @@
+import { parse, TSESTree } from '@typescript-eslint/typescript-estree';
+import { readFileSync } from 'node:fs';
+
+interface SuperClass {
+    name: string;
+    path: string;
+    superConstructorCalls: { arguments: string[] }[];
+    childClasses: ChildClass[];
+}
+
+interface ChildClass {
+    superClass: string;
+    name: string;
+    memberVariables: Map<string, MemberVariable>;
+    parentMethodCalls: ParentMethodCalls[];
+}
+
+interface ParentMethodCalls {
+    name: string;
+    parameters: MemberVariable[];
+}
+
+interface MemberVariable {
+    name: string;
+    type: string;
+    value?: string;
+}
+
+export class Preprocessor {
+    public static PREPROCESSING_RESULTS = new Map<string, SuperClass>();
+    public static readonly pathPrefix = ''
+    private readonly directoryPrefix = 'src/main/webapp/';
+    private readonly fileToPreprocess: string;
+    private ast: TSESTree.Program;
```

Comment:
Follow-up: Consider reusing this for the post processor, so that it needs to be generated only once.